Mathlib/Topology/Support.lean

```diff
@@ -110,6 +110,18 @@ theorem mulTSupport_mul [TopologicalSpace X] [Monoid α] {f g : X → α} :
     ((mulSupport_mul f g).trans (union_subset_union (subset_mulTSupport _) (subset_mulTSupport _)))
     (isClosed_closure.union isClosed_closure)
 
+theorem tsupport_smul_left [TopologicalSpace X] [Semiring R] [AddCommMonoid M] [Module R M]
+    [NoZeroSMulDivisors R M] (f : X → R) (g : X → M) : tsupport (f • g) ⊆ tsupport f := by
+  apply closure_mono
+  erw [support_smul]
+  exact inter_subset_left _ _
+
+theorem tsupport_smul_right [TopologicalSpace X] [Semiring R] [AddCommMonoid M] [Module R M]
+    [NoZeroSMulDivisors R M] (f : X → R) (g : X → M) : tsupport (f • g) ⊆ tsupport g := by
```

Comment:
Could you move this next to [tsupport_smul_subset_left](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Support.html#tsupport_smul_subset_left), use the same naming conventions, and weaken the hypotheses to those required by [Function.support_smul_subset_right](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Module/Basic.html#Function.support_smul_subset_right) (plus topology of course) ?

Reply:
I tried, but for some reason things don't unify.

Reply:
Ah yes, because `support_smul_subset_right` is about multiplying by a constant "scalar". Would you mind renaming it to `support_const_smul_subset` (thus adjusting the `#align`), and adding
```lean
lemma support_smul_subset_right [Zero M] [SMulZeroClass R M] (f : α → R) (g : α → M) :
    support (f • g) ⊆ support g := fun x hbf hf ↦ hbf <| by rw [Pi.smul_apply', hf, smul_zero]
```
right after [tsupport_smul_subset_left](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Support.html#tsupport_smul_subset_left)?

Reply:
Indeed. Just fixed the archive.